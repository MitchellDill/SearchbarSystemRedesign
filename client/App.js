import React from 'react';
import Axios from 'axios';
import DropDown from './dropDown';


//when you enter instead of clicking on an  Autofilling from the drop down menu
//then get the id for the top   Autofilling which will be this.state  Autofilling[0]
class Search extends React.Component{
    constructor(props){
        super(props);
        this.state ={
            input: '', //current state of search bar
            items: [], //names of items 
            Autofilling: [], //items that will be render below the search bar
            showMenu: true //maybe new component to control the menu sropdown 
        }
        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleChange(e){
        e.preventDefault();
        //need to compare realtime keystrokes to filter thru names
        // const names = this.state.items.map(  Autofilling) =>  Autofilling.name);
        const names = this.state.items; //taking names that are in the state currently
        let autoFill = [];
        let searching = [];
        names.filter((name) => { //only works when the word after the initial word is directly after in the Autofilling name
            if(name.includes(e.currentTarget.value.toLowerCase())){ //try to compare by each word, and if it isnt included then drop it 

                if(e.currentTarget.value.length !== 0){
                    
                    autoFill.push(name);
                    //************************************************* */check to see which. Autofilling has more in common with the input
                    this.setState({ 
                        input: e.currentTarget.value,
                        Autofilling: autoFill 
                    });
                }else{
                    this.setState({ Autofilling: [] })                                                                 
                }
            }
        })
    }

    handleSubmit(){
        console.log(this.state.input);
        Axios.post('http://localhost:3001/find', { name: this.state.Autofilling[0]}) //going to use the first arr in the autocorrection
        .then(response => {
            window.dispatchEvent(new CustomEvent('productChanged', {
                detail: {
                  id: response.data[0].productID
                }
              }));
            // console.log(response.data[0].productID)) //retrieves the id of the first itek in the autocorrect list
        })
        .catch(err => console.log(err))
    }

    componentWillMount(){
        Axios.get('http://localhost:3001/items') //recieving all names from all the Autofilling names
        .then(response => { this.setState({ items: response.data }) })
        .catch(err => console.log('ERR', err));
    }

    filterThruAutoFill(){
        if(AutoFilling.length > 0){
            console.log("filter thru more with input");
        }
    }

    render(){
        return(
            <div>
 
                {console.log(this.state.input)}
                {/* <nav className="white" role="navigation">
                
                    <div className="nav-wrapper container row">
                    <div className="col s2"><img src="../Zbay.png"></img></div>
                    <div className="col s3"><input type= "text" placeholder="Search" onChange={this.handleChange}></input></div>
                    <div><button onClick={this.handleSubmit}>Go</button></div> */}

            
                    {/* <div className="col s1"><button class="btn waves-effect waves-light" type="submit"  name="action">Submit
                        <i class="material-icons right">send</i>
                    </button></div> */}
        
                    {/* <div className="col s1"></div>
                        <ul className="left hide-on-med-and-down">
                        <li><a href="#">Navbar Link</a></li>
                        </ul>
                
                        <ul id="nav-mobile" className="sidenav">
                        <li><a href="#">Navbar Link</a></li>
                        </ul>
                        <a href="#" data-target="nav-mobile" className="sidenav-trigger"><i class="material-icons">menu</i></a>
                    </div>
                </nav> */}
                <div className="w3-white">
                <img src="../Zbay.png"></img>
                <div className="w3-dropdown-hover">
                    <button className="w3-button">Shop by category</button>
                    <div className="w3-dropdown-content">
                    <table>
                        <tr>
                            <th>Collectibles and art</th>
                            <th>Fashion</th>
                        </tr>

                        <tr>
                            <td>Collectibles</td>

                        </tr>
                        <tr>
                            <td>Coins & paper money Antiques</td>
                        
                        </tr>

                        <tr>
                            <td>Sports memorabillia</td>

                        </tr>
                    </table>
                    </div>
                </div>
                
                
                <input type="text" className="w3-bar-item w3-button" placeholder="Search for anything" onChange={this.handleChange}></input>
                 <button className="w3-bar-item w3-button w3-blue" onClick={this.handleSubmit}>Search</button>
                
                </div>
                

                {/* <form class="form-inline md-form form-sm mt-0">
                <i class="fas fa-search" aria-hidden="true"></i>
                <input Style="padding:0 0 0 20px" class="form-control form-control-sm ml-3 w-75" type="text" placeholder="Search"
                    aria-label="Search" onChange={this.handleChange}/>
                </form> */}

                
                {(this.state.Autofilling.length !== 0) ? 
                <div>
                <p>{this.state.Autofilling[0]}</p>
                <p>{this.state.Autofilling[1]}</p>
                <p>{this.state.Autofilling[2]}</p>
                <p>{this.state.Autofilling[3]}</p>
                </div>
                : null
                }
            </div>
        )
    }
}

export default Search;