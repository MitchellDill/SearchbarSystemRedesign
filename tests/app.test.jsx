import React from "react";
import renderer from "react-test-renderer";
import App from "../client/App";

it.skip("coninues to render App correctly", () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
