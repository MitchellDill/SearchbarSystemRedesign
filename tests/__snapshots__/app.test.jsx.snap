// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coninues to render App correctly 1`] = `
<div>
  <div
    className="top-bar"
  >
    <ul
      className="top"
    >
      <li
        className="user"
      >
        <span
          className="span-user"
        >
          Hi 
          <b>
            User
          </b>
           (
          <a
            className="signin"
            href="#"
          >
            Sign in
          </a>
          )
           
        </span>
      </li>
      <li
        className="Daily-Deals"
      >
        <span
          className="span-dailydeals"
        >
          Daily Deals
        </span>
      </li>
      <li
        className="Gift-Cards"
      >
        <span
          className="span-giftcards"
        >
          Gift Cards
        </span>
      </li>
      <li
        className="Help-Contact"
      >
        <span
          className="span-help"
        >
          Help & Contact
        </span>
      </li>
      <li
        className="deals"
      >
        <img
          id="deals"
          onClick={[Function]}
          src="https://ir.ebaystatic.com/cr/v/c1/61203_071519__GG_SM_HRZ_RW29_GenericPrimeMsg_Doodle_150x30_R1.gif"
        />
      </li>
    </ul>
    <ul
      id="right-top"
    >
      <li
        className="sell"
      >
        <span
          className="span-sell"
        >
          sell
        </span>
      </li>
      <li
        className="my-ebay"
      >
        <span
          className="span-My"
        >
          My ZBay
        </span>
      </li>
      <li
        className="notifications"
      >
        <button
          className="bell-icon"
        >
          <i
            id="alert"
          />
        </button>
      </li>
      <li
        className="Search-cart"
      >
        <a
          className="cart-icon"
          onClick={[Function]}
        >
          <i
            id="shop"
          />
        </a>
      </li>
    </ul>
  </div>
  <div
    className="Search-component"
  >
    <div
      className="w4-white"
    >
      <img
        id="Search-Zbay-logo"
        onClick={[Function]}
        src="https://searchcomponent.s3.us-east-2.amazonaws.com/searchpngs/Zbay.png"
      />
      <div
        className="w3-dropdown-hover"
      >
        <button
          aria-expanded="false"
          className="category-dropdown"
        >
          Shop by
           
          <p
            rowSpan="3"
          >
            category
            <i
              id="arrow-dropdown"
            />
          </p>
        </button>
        <div
          className="w3-dropdown-content"
        >
          <table
            id="drop"
          >
            <tbody>
              <tr>
                <th>
                  Collectibles and art
                   
                  <i
                    className="right"
                    id="arrow"
                  />
                </th>
                <th>
                  Fashion
                  <i
                    className="right"
                    id="arrow"
                  />
                </th>
                <th>
                  Sporting goods
                  <i
                    className="right"
                    id="arrow"
                  />
                </th>
              </tr>
              <tr>
                <td>
                  Collectibles
                </td>
                <td>
                  Women
                </td>
                <td>
                  Outdoor sports
                </td>
              </tr>
              <tr>
                <td>
                  Coins & paper money Antiques
                </td>
                <td>
                  Men
                </td>
                <td>
                  Team sports
                </td>
              </tr>
              <tr>
                <td>
                  Antiques
                </td>
                <td>
                  Watches
                </td>
                <td>
                  Exercise & fitness
                </td>
              </tr>
              <tr>
                <td>
                  Sports memorabillia
                </td>
                <td>
                  Shoes
                </td>
                <td>
                  Golf
                </td>
              </tr>
              <tr>
                <th
                  rowSpan="1"
                >
                  Electronics
                  <i
                    className="right"
                    id="arrow"
                  />
                </th>
                <th
                  rowSpan="1"
                >
                  Home & Garden
                  <i
                    className="right"
                    id="arrow"
                  />
                </th>
                <th
                  rowSpan="1"
                >
                  Toys & Hobbies
                  <i
                    className="right"
                    id="arrow"
                  />
                </th>
              </tr>
              <tr>
                <td>
                  Computers & tables
                </td>
                <td>
                  Yard, garden & outdoor
                </td>
                <td>
                  Vintage & anitques toys
                </td>
              </tr>
              <tr>
                <td>
                  Cameras & photos
                </td>
                <td>
                  Crafts
                </td>
                <td>
                  Kids toys
                </td>
              </tr>
              <tr>
                <td>
                  TV, audio & surveillance
                </td>
                <td>
                  Home improvement
                </td>
                <td>
                  Action figures
                </td>
              </tr>
              <tr>
                <td>
                  Cell phones & accessories
                </td>
                <td>
                  Pet supplies
                </td>
                <td>
                  Dolls & bears
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <input
        className="Searchbar"
        id="search"
        list="item-names"
        onChange={[Function]}
        onKeyPress={[Function]}
        placeholder="Search for anything"
        type="text"
        value=""
      />
      <datalist
        id="item-names"
      >
        <select />
      </datalist>
      <select
        className="dropdown"
      >
        <option
          value="All Categories"
        >
          All Categories
        </option>
        <option
          value="Antiques"
        >
          Anitques
        </option>
        <option
          value="Ar"
        >
          Art
        </option>
        <option
          value="Baby"
        >
          Baby
        </option>
        <option
          value="Books"
        >
          Books
        </option>
        <option
          value="Business & Industrial"
        >
          Business & Industrial
        </option>
        <option
          value="Cameras & photos"
        >
          Cameras & photos
        </option>
        <option
          value="Cell phone & accessories"
        >
          Cell phone & accessories
        </option>
      </select>
      <button
        className="Search"
        onClick={[Function]}
      >
        Search
      </button>
      <div
        className="showcontent"
      />
      <div
        className="T2"
      >
        <hr />
        <div
          className="table2"
        >
          <table
            className="nav"
          >
            <tbody>
              <tr>
                <td>
                  Home
                </td>
                <td>
                  Saved
                </td>
                <td>
                  Motors
                </td>
                <td>
                  Fashion
                </td>
                <td>
                  Electronics
                </td>
                <td>
                  Collectibles & Art
                </td>
                <td>
                  Home & Garden
                </td>
                <td>
                  Sporting Goods
                </td>
                <td>
                  Toys
                </td>
                <td>
                  Business & industrial
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;
